global_planner_node:
  ros__parameters:
    # enable the local minimum curvature optimisation
    enable_opt: true

    # enable profiling
    enable_profiler: true

    # enable visualisation
    enable_visualisation: true

    # ROS TOPICS
    path_subscriber_topic: "/estimation/final_path"
    VE_subscriber_topic: "/vcu_msgs/velocity_estimation"
    global_reference_publisher_topic: "/planning/global_reference"
    global_reference_viz_publisher_topic: "/planning/viz/global_reference"
    global_flag_publisher_topic: "/planning/global_reference/status"
    estimation_subscriber_topic: "/estimation/online_map"
    log_publisher_topic: "/planning/global_reference/log"

    # The first point of the path must be fixed on the middle line
    first_point_fixed: false

    # The last point of the path must be fixed on the middle line
    last_point_fixed: false

    # The path is closed
    closed_path: true

    # Enable safe set learning
    enable_safe_set_learning: false

    # safe velocity multiplier (safe_velocity_at_point = safe_velocity_multiplier * velocity_at_point)
    safe_velocity_multiplier: 0.7

    # safe velocity blending factor (safe_velocity_at_point = blend * driven_velocity + (1 - blend) * ay-curvature_safe_velocity)
    safe_velocity_blend: 0.2

    # The width of the car
    car_width: 1.5

    # The safety distance on each side
    safety_margin: 0.65

    # Number of iterations for the optimisation
    nb_iter: 35

    # frequency at which the global map in local frame is published by a timer
    # if set to 0, the global map is published only along a callback to SLAM (i.e. when a new transform is available)
    global_planner_frequency: 40.0

    # Length of the trak in meters
    track_length: 287

    # Distance between two points of the path in meters
    points_spacing: 0.5

    # Radius of the path to cut and publish
    path_radius: 30.0

    # Maximum lateral acceleration used to compute the speed
    max_lateral_acceleration: 0.15

    # Minimum longitudinal speed
    min_longitudinal_speed: 0.001

    # Maximum longitudinal speed used to compute the speed
    max_longitudinal_speed: 4.0

    # Number of points around each point to consider for curvature smoothing
    curvature_smoothness: 30

    # Standard deviation of the gaussian kernel used for curvature smoothing
    gaussian_kernel_std: 5.0

    # Maximum curvature threshold to consider the point as feasible
    max_curvature: 0.5

    # Maximum distance between two points to consider the path as feasible
    max_distance: 9.0

    # Substracted curvature to shift lateraly the speed profile
    kappa_shift: 0.0

    # HPIPM parameters (more details in the min_curvature_opt.hpp)
    mode: 1
    iter_max: 50
    alpha_min: 0.000001
    mu0: 10.0
    tol_stat: 0.000001
    tol_eq: 0.000001
    tol_ineq: 0.000001
    tol_comp: 0.000001
    reg_prim: 0.000001
    reg_dual: 0.000001
    warm_start: 0
    pred_corr: 1
    split_step: 1
